### PR TITLE
[docs][python] Improve documentation for xbmc.sleep and xbmc.Monitor(…

### PR DESCRIPTION
**Pull request by enen92**
…).waitForAbort

## Description
The note provided in the `xbmc.sleep()` method doesn't make sense anymore since `xbmc.Monitor().waitForAbort()` was introduced and actually encourages addon devs to do it wrong.
We have been suggesting during reviews to only use `xbmc.sleep()` in cases where the addon needs to sleep for really small ammounts of time - to avoid Kodi hanging on shutdown. For long periods of time, a `Monitor` (and `waitForAbort()`) should be used instead.

## Motivation and Context
Improve docs

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [x] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [x] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Review by ronie — APPROVED** [review]
perfect, and well explained in the docs.

**Comment by enen92** [comment]
Made just a small change to avoid being misunderstood as "using xbmc.sleep for 5 seconds is fine". Ideally this should only be used in the milisecond range.
If you sleep only for 4 seconds and the logic that follows the sleep call takes also a few seconds to exit, it'll also hang kodi.

File: xbmc/interfaces/legacy/ModuleXbmc.h [hunk 1]
```diff
@@ -212,14 +212,22 @@ namespace XBMCAddon
...
+    /// Sleeps for 'time' (msec).
+    /// \anchor xbmc_Sleep
     ///
     /// @param time                 integer - number of msec to sleep.
     ///
     /// @throws PyExc_TypeError     If time is not an integer.
     ///
-    /// @note This is useful if you have for example a Player class that is
-    ///       waiting for onPlayBackEnded() calls.
+    /// @warning This is usefull if you need to sleep for a small ammount of time
```
Comment (MilhouseVH) [inline xbmc/interfaces/legacy/ModuleXbmc.h:222]:
```suggestion
    /// @warning This is useful if you need to sleep for a small amount of time
```
Typos

**Comment by enen92** [comment]
jenkins build and merge

**Review by basrieter — APPROVED** [review]

**Comment by basrieter** [comment]
Makes good sense to me. Description is clear.

File: xbmc/interfaces/legacy/ModuleXbmc.h [hunk 2]
```diff
@@ -212,14 +212,22 @@ namespace XBMCAddon
...
-    /// @note This is useful if you have for example a Player class that is
-    ///       waiting for onPlayBackEnded() calls.
+    /// @warning This is useful if you need to sleep for a small amount of time
+    /// (milisecond range) somewhere in your addon logic. Please note that Kodi
+    /// will attempt to stop any running scripts when signaled to exit and wait for a maximum
+    /// of 5 seconds before trying to force stop your script. If your addon makes use
+    /// of \ref xbmc_Sleep "xbmc.sleep()" incorrectly (long periods of time, e.g. that exceed
+    /// the force stop waiting time) it may lead to Kodi hanging on shutdown.
+    /// In case your addon needs long sleep/idle periods use
+    /// \ref xbmc_Monitor_waitForAbout "xbmc.Monitor().waitForAbort(secs)"
```
Comment (MrTarantula) [inline xbmc/interfaces/legacy/ModuleXbmc.h:229]:
```suggestion
    /// \ref xbmc_Monitor_waitForAbort "xbmc.Monitor().waitForAbort(secs)"
```

File: xbmc/interfaces/legacy/Monitor.h [hunk 3]
```diff
@@ -257,6 +257,7 @@ namespace XBMCAddon
       /// @brief \python_func{ waitForAbort([timeout]) }
       ///-----------------------------------------------------------------------
       /// Wait for Abort
+      /// \anchor xbmc_Monitor_waitForAbout
```
Comment (MrTarantula) [inline xbmc/interfaces/legacy/Monitor.h:260]:
```suggestion
      /// \anchor xbmc_Monitor_waitForAbort
```

**Comment by enen92** [comment]
jenkins build and merge

**Pull request merged**